### PR TITLE
STEP01_TDD로 개발하기 PR 제출 합니다.

### PR DESCRIPTION
**Pull request by wldyd2214**
아래의 항목들에 대해서 코드 리뷰를 받고 싶습니다.
  1. userPointUse 메소드에서 부터 호출되는 userPointUsedProcess 비지니스 로직
  2. 컨트롤러 테스트 항목들에 대한 코드
  3. 서비스 테스트 항목들에 대한 코드
  4. 레포지토리 테스트 항목들에 대한 코드

### CODE REVIEW

File: src/test/java/io/hhplus/tdd/api/service/point/PointHistoryServiceTest.java [hunk 1]
```diff
@@ -23,6 +23,8 @@ class PointHistoryServiceTest {
     @Autowired
     private PointHistoryService pointHistoryService;
 
+    // TODO: 히스토리 서비스 테스트 항목들에 대한 코드 리뷰를 받고 싶습니다.
+
     @DisplayName("특정 유저의 포인트 충전/이용 내역을 조회한다.")
```
Comment (hubtwork) [inline src/test/java/io/hhplus/tdd/api/service/point/PointHistoryServiceTest.java]: 확인하고자 하는 것은, 동일한 유저에 대한 History 가 잘 쌓여있는가 ? 인데, 2번 유저에 대한 given 이 필요할까요 ?

File: src/main/java/io/hhplus/tdd/api/service/point/PointService.java [hunk 2]
```diff
@@ -38,6 +38,7 @@ private UserPoint chargeUserPointProcess(long id, long amount) {
         return insertUserPointOrUpdate(id, sumPoint);
     }
 
+    // TODO: userPointUse 메소드에서 부터 호출되는 userPointUsedProcess 비지니스 로직의 클린 코드 리뷰를 받고 싶습니다.
```
Comment (hubtwork) [inline src/main/java/io/hhplus/tdd/api/service/point/PointService.java]: 불필요하게 private 으로 함수를 다 쪼개놓아서 오히려 테스트 / 가독성 측면에서 좋지 않은 방향성으로 보입니다. 
또한 amount 도 음수이고, user 의 포인트 또한 음수인 경우 ? validation 을 통과해 포인트를 마치 충전하는 것과 같은 calculation 이 일어날 텐데, 이에 대한 고려가 부족합니다.

File: src/test/java/io/hhplus/tdd/api/controller/point/PointControllerTest.java [hunk 3]
```diff
@@ -33,6 +33,8 @@ class PointControllerTest {
     @MockBean
     protected PointHistoryService pointHistoryService;
 
+    // TODO: 컨트롤러 테스트 항목들에 대한 코드 리뷰를 받고 싶습니다.
```
Comment (hubtwork) [inline src/test/java/io/hhplus/tdd/api/controller/point/PointControllerTest.java]: 컨트롤러 단위 테스트의 경우, `jsonPath` 로 각 항목들을 점검하게 되면 개발자의 실수에 의해 누락값이 존재할 수 있고 누락된 경우, 나머지 검증값이 같다면 테스트 통과로 이어지는 불상사가 발생해요 ( Edge Case 가 존재하지만 테스트는 통과해 서비스의 오류로 이어질 수 있음 )

File: src/test/java/io/hhplus/tdd/api/service/point/PointHistoryServiceTest.java (hunk 1 above)
Comment (wldyd2214) [reply src/test/java/io/hhplus/tdd/api/service/point/PointHistoryServiceTest.java]: 여러 데이터 사이에서 특정 유저의 데이터만 잘 가져오는지를 확인하면 좀 더 신뢰성이 있을 것 같아 2번 유저에 대한 given을 추가했었습니다.
